### PR TITLE
chore: refactoring handle error

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Adds a new service handle error.
* Updates record service with the new handle error.

### CODE REVIEW

File: projects/rero/ng-core/src/lib/record/record.handle-error.service.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    // If not, then return default message
+    switch (error.status) {
+      case 400:
+        return throwError({ status: error.status, title: this.translateService.instant('Bad request') });
+      case 401:
+        return throwError({ status: error.status, title: this.translateService.instant('Unauthorized') });
+      case 403:
+        return throwError({ status: error.status, title: this.translateService.instant('Forbidden') });
+      case 404:
+        return throwError({ status: error.status, title: this.translateService.instant('Not found') });
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/record.handle-error.service.ts:75]: Do these message should contains `server`?

File: projects/rero/ng-core/src/lib/record/record.handle-error.service.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+   * @param translateService - TranslateService
+   */
+  constructor(protected translateService: TranslateService) { }
+
+  /**
+   * handle http error
+   * @param error - HttpErrorResponse
+   * @returns Observable
+   */
+  handleError(error: HttpErrorResponse, resourceName?: string): Observable<never> {
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/record.handle-error.service.ts:37]: resourceName seems not used.

File: projects/rero/ng-core/src/lib/record/record.service.ts [hunk 3]
```diff
@@ -360,48 +363,19 @@ export class RecordService {
...
-      case 401:
-        return throwError({ status: error.status, title: this._translateService.instant('Unauthorized') });
-      case 403:
-        return throwError({ status: error.status, title: this._translateService.instant('Forbidden') });
-      case 404:
-        return throwError({ status: error.status, title: this._translateService.instant('Not found') });
-      default:
-        return throwError({ status: 500, title: this._translateService.instant('An error occurred') });
-    }
+  private _handleError(error: HttpErrorResponse, resourceName?: string): Observable<never> {
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/record.service.ts:377]: `resourceName` can be removed.

**Review by jma — APPROVED** [review]

File: projects/rero/ng-core/src/lib/record/record.handle-error.service.ts (hunk 2 above)
Comment (Garfield-fr) [reply projects/rero/ng-core/src/lib/record/record.handle-error.service.ts:37]: This parameter is used in the UI. It's to have the same signature.

File: projects/rero/ng-core/src/lib/record/record.service.ts (hunk 3 above)
Comment (Garfield-fr) [reply projects/rero/ng-core/src/lib/record/record.service.ts:377]: This parameter is used in the UI. It's to have the same signature.

**Review by zannkukai — APPROVED** [review]

**Pull request merged**